### PR TITLE
Phil spel num diff

### LINKED ISSUE #23: Integration Aufgabe
Edit by Philipp:

- [x] MoxUnit did not work on my Octave, only offline after installing it manually using the instructions on https://github.com/MOxUnit/MOxUnit
- [x] is it necessary to check the stepsize `df`?

### LINKED ISSUE #146: Differentiation exercise
add an introduction to difference quotients

### PR DESCRIPTION
**Pull request by PhiSpel**
created an exercise about difference quotients

closes #23

closes #146

### CODE REVIEW

File: content/06_Differentiation_Integration/02_integration_and_stepcontrol.ipynb [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+   "outputs": [],
+   "source": [
+    "test_numInt()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Den Quellcode des `unit tests` können Sie hier ausklappen. Sehen Sie sich [](content:references:unit_testing_intro) an, um Ihre Erinnerung über Unit Tests aufzufrischen.\n",
```
Comment (sandro-elsweijer) [inline content/06_Differentiation_Integration/02_integration_and_stepcontrol.ipynb:66]: Grammar

**Review by sandro-elsweijer — COMMENTED** [review]
Only minor things need to be done

File: content/06_Differentiation_Integration/02_integration_and_stepcontrol.ipynb (hunk 1 above)
Comment (PhiSpel) [reply content/06_Differentiation_Integration/02_integration_and_stepcontrol.ipynb]: see d5197fb

File: content/06_Differentiation_Integration/01_differentiation.ipynb [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+   "source": [
+    "## Aufgabe 2\n",
+    "\n",
+    "- Lagern Sie Ihren Test so aus, dass Sie für beliebige Funktionen die eingegebene analytische Lösung mit der numerischen Lösung vergleichen können.\n",
+    "- Werten Sie anstatt für diskrete Werte für ganze Intervalle die durchschnittliche und die maximale Abweichung `meandiff`, `maxdiff` von der analytischen Lösung aus.\n",
+    "- Werten Sie die benötigte Zeit `time` und die Anzahl an Funktionsauswertungen `n` aus.\n",
+    "- Verringern Sie die Anzahl der Funktionsauswertungen, indem Sie diese wiederverwenden.\n",
+    "\n",
+    "```{admonition} Hinweis\n",
+    "Bei sehr komplexen Funktionen ist die Verringerung von Funktionsauswertungen entscheident für die Performanz Ihres Programms.\n",
```
[the remark this answers is not in the document]
Comment (PhiSpel) [reply content/06_Differentiation_Integration/01_differentiation.ipynb]: see d5197fb

File: content/06_Differentiation_Integration/01_differentiation.ipynb [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "428d6916",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%file numDiff2.m\n",
```
[the remark this answers is not in the document]
Comment (PhiSpel) [reply content/06_Differentiation_Integration/01_differentiation.ipynb]: see d5197fb

File: content/06_Differentiation_Integration/01_differentiation.ipynb [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f2540c5d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%file numDiff.m\n",
```
[the remark this answers is not in the document]
Comment (PhiSpel) [reply content/06_Differentiation_Integration/01_differentiation.ipynb]: Good point. I removed the keyword so the code will be executed directly. Creating a file is not necessary.
see d5197fb

File: content/06_Differentiation_Integration/01_differentiation.ipynb [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+   "source": [
+    "# Numerische Differentiation\n",
+    "\n",
+    "Um Ableitungen von Funktionen zu approximieren, die wir analytisch nicht berechnen können oder wollen, verwenden wir **Differenzenquotienten**. Im Wesentlichen unterscheiden wir zwischen\n",
+    "\n",
+    "- *Vorwärtsdifferenzen*, die für die Ableitung am Punkt x Funktionswerte größer als x mit einbeziehen, zum Beispiel $f'(x) \\approx \\frac{f(x+dx) - f(x)}{dx}$,\n",
+    "- *Zentrale Differenzen*, die Funktionswerte größer und kleiner als x mit einbeziehen, zum Beispiel $f'(x) \\approx \\frac{f(x+dx) - f(x-dx)}{2 dx}$, und\n",
+    "- *Rückwärtsdifferenzen*, die Funktionswerte kleiner als x mit einbeziehen, zum Beispiel $f'(x) \\approx \\frac{f(x) - f(x-dx)}{dx}$.\n",
+    "\n",
+    "Differenzenquotienten können für beliebig hohe Ableitungen gebildet werden. Der Fehler wird mit geringerer Schrittweite $dx$ kleiner. Wie stark der Fehler kleiner wird, beschreibt die Fehlerordnung $\\mathcal{O}$. Zum Beispiel ist für quadratische Fehlerordnung $\\mathcal{O}(dx^2)$, also $dx_2 = \\frac{dx_1}{2} \\rightarrow \\epsilon_2 = \\frac{\\epsilon_1}{2^2}$.\n",
```
[the remark this answers is not in the document]
Comment (PhiSpel) [reply content/06_Differentiation_Integration/01_differentiation.ipynb]: see d5197fb

**Review by sandro-elsweijer — CHANGES_REQUESTED** [review]

**Review by sandro-elsweijer — APPROVED** [review]

**Pull request merged**